File: MotivApp/app/src/main/java/com/motiv/piotr/dao/LinksDao.java
package com.motiv.piotr.dao;

import androidx.room.*;
import com.motiv.piotr.LinksWithReferences;

@Dao
public interface LinksDao {

    @Insert
    public long saveLinks(com.motiv.piotr.Links links);

    @Delete
    public int deleteLinks(com.motiv.piotr.Links links);

    @Query("SELECT * FROM links")
    @Transaction
    public java.util.List<LinksWithReferences> getLinkss();

    @Query("SELECT * FROM links WHERE id = :id")
    @Transaction
    public LinksWithReferences getLinks(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void saveLinkss(java.util.List<com.motiv.piotr.Links> links);
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/LinksDao.kt
 
package com.motiv.piotr.dao
import androidx.room.*
import com.motiv.piotr.Links
import com.motiv.piotr.LinksWithReferences
import kotlin.collections.List
@Dao
public interface LinksDao {

    @Insert
    public fun saveLinks(links: com.motiv.piotr.Links): Long@Delete
    public fun deleteLinks(links: com.motiv.piotr.Links): Int@Query("SELECT * FROM links")
    @Transaction
    public fun getLinkss(): List<LinksWithReferences>@Query("SELECT * FROM links WHERE id = :id")
    @Transaction
    public fun getLinks(id: String): LinksWithReferences@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun saveLinkss(links: List<com.motiv.piotr.Links>): Unit
}
